### PR TITLE
feat: quality analytics pipeline

### LINKED ISSUE #149: Quality analytics pipeline: track SKIP rates, parse failures, and model performance
## Problem

We are manually combing through consumer repos to find SKIPs, parse failures, and bad reviews. This doesn't scale. We need automated quality tracking to iterate scientifically on review quality.

## What To Track

Per review run:
- **SKIP rate** — percentage of reviewers that SKIPped
- **Parse failure rate** — percentage that produced output but couldn't be parsed
- **Model used** — which model produced which outcome
- **Verdict distribution** — PASS/WARN/FAIL/SKIP by model and perspective
- **Runtime** — seconds per reviewer, by model
- **Confidence scores** — distribution by model and perspective
- **Findings quality** — severity distribution, hallucination markers

### Option A: GitHub Actions job summary + artifact (MVP)
- Add a `quality-report.json` artifact to each council verdict run
- Contains all metrics above
- Periodically aggregate via a cron workflow on the cerberus repo
- Cheap, no infra, works today

### Option B: Webhook to analytics service
- Council verdict step POSTs metrics to an endpoint
- PostHog, or a simple Cloudflare Worker → D1 store
- Enables dashboards and alerting
- Requires consumer to opt-in (new action input)

### Option C: GitHub Issue-based tracking
- On SKIP or parse-failure, auto-file an issue on the cerberus repo with structured labels
- Can query with `gh issue list --label "quality/skip"`
- Zero infra but noisy

### Recommendation

Start with **Option A** (artifact-based). Graduate to **Option B** when we have enough data to justify dashboards. Skip Option C (too noisy).

## Acceptance Criteria

- [ ] Each council verdict run produces a `quality-report.json` artifact
- [ ] Report includes: repo, PR, SHA, per-reviewer model, verdict, confidence, runtime, parse outcome
- [ ] Aggregation script can be run locally: `python3 scripts/quality-report.py --repo misty-step/moneta --last 20`
- [ ] Script produces summary: SKIP rate, model rankings, average runtime

## Related
- #141 (SENTINEL reliability — this gives us data to evaluate fixes)
- #148 (model pool — this gives us data to evaluate model performance)
- #133 (eval system — complementary, that's synthetic eval, this is production telemetry)

### PR DESCRIPTION
**Pull request by kaylee-mistystep**
## Summary

Implements the quality analytics pipeline from #149 to track SKIP rates, parse failures, and model performance.

### 1. Quality Report Generation (scripts/aggregate-verdict.py)
- New generate_quality_report() function creates quality-report.json
- Per-reviewer metrics: verdict, confidence, runtime, model used, fallback status, parse failure status
- Per-model aggregation: count, verdict distribution, runtime stats, skip rate, parse failure rate
- Overall summary: skip rate, parse failure rate, council verdict distribution

### 2. Artifact Upload (verdict/action.yml)
- Uploads quality-report.json as artifact with 30-day retention
- Added GH_PR_NUMBER and GITHUB_REPOSITORY env vars for metadata

### 3. Aggregation Script (scripts/quality-report.py)
- CLI tool: python3 scripts/quality-report.py --repo misty-step/moneta --last 20
- Outputs human-readable summary with model rankings
- Fetches artifacts from recent CI runs via gh CLI

## Acceptance Criteria

- [x] Each council verdict run produces a quality-report.json artifact
- [x] Report includes repo, PR, SHA, per-reviewer metrics
- [x] Aggregation script works locally
- [x] Script produces SKIP rate, model rankings, average runtime

Closes #149

## Summary by CodeRabbit

* **New Features**
  * Added quality report generation capturing reviewer details, per-model statistics, verdict distribution, and performance metrics.
  * New tool to aggregate quality reports from CI runs or local artifacts and produce JSON or human-readable summaries.

* **Tests**
  * Added tests validating report structure, model aggregates, rates, and edge cases (empty inputs, skipped artifacts).

* **Chores**
  * CI workflow updated to upload generated quality report artifacts.

### CODE REVIEW

File: tests/test_quality_report.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+"""Test the quality report generation."""
+import json
+import sys
+from pathlib import Path
+
+# Add scripts to path
+sys.path.insert(0, str(Path(__file__).parent.parent / "scripts"))
+
+# Import directly
+exec(open(Path(__file__).parent.parent / "scripts" / "aggregate-verdict.py").read())
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply tests/test_quality_report.py]: Fixed — now uses `importlib.util.spec_from_file_location` instead of `exec()`.

File: scripts/quality-report.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+def run_gh(args: list[str]) -> str:
+    """Run gh CLI and return stdout."""
+    result = subprocess.run(
+        ["gh"] + args,
+        capture_output=True,
+        text=True,
+    )
+    if result.returncode != 0:
+        print(f"gh error: {result.stderr}", file=sys.stderr)
+        sys.exit(1)
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply scripts/quality-report.py]: Fixed — `run_gh` now raises `GHError` instead of calling `sys.exit(1)`. Callers catch `GHError`.

File: tests/test_quality_report.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# Validate structure
+assert "meta" in report
+assert "summary" in report
+assert "reviewers" in report
+assert "models" in report
+
+assert report["meta"]["repo"] == "misty-step/test"
+assert report["summary"]["total_reviewers"] == 3
+assert report["summary"]["skip_count"] == 1
+assert report["summary"]["skip_rate"] == 1/3
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply tests/test_quality_report.py]: Both the implementation and test use `round(1/3, 4)`, so the comparison is exact. No floating-point issue here since both sides are rounded identically.

File: scripts/aggregate-verdict.py [hunk 4]
```diff
@@ -195,6 +197,123 @@ def aggregate(verdicts: list[dict], override: Override | None = None) -> dict:
...
+        stats["avg_runtime_seconds"] = stats["total_runtime_seconds"] / count if count > 0 else 0
+        stats["median_runtime_seconds"] = statistics.median(runtimes) if runtimes else 0
+        stats["skip_rate"] = stats["verdicts"]["SKIP"] / count if count > 0 else 0
+        stats["parse_failure_rate"] = stats["parse_failures"] / count if count > 0 else 0
+        stats["fallback_rate"] = stats["fallback_count"] / count if count > 0 else 0
+
+    # Verdict distribution
+    verdict_distribution = {"PASS": 0, "WARN": 0, "FAIL": 0, "SKIP": 0}
+    for v in verdicts:
+        verdict_distribution[v["verdict"]] += 1
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply scripts/aggregate-verdict.py]: Fixed in 1b367ee — unexpected verdicts now create a new key instead of KeyError.
